client/src/AppBundle/Controller/Admin/StatsController.php

```diff
@@ -49,6 +58,106 @@ public function statsAction(Request $request, ReportSubmissionSummaryMapper $map
+     */
+    public function downloadSpreadsheet($spreadsheet)
+    {
+        $extension = 'Xlsx';
+        $fileName = sprintf('Satisfaction_%s.xlsx', date('YmdHi'));
+        $writer = IOFactory::createWriter($spreadsheet, $extension);
+        header('Content-Type: application/vnd.openxmlformats-officedocument.spreadsheetml.sheet');
+        header("Content-Disposition: attachment; filename=\"{$fileName}\"");
+        $writer->save('php://output');
+        exit();
+    }
+
+    /**
+     * @param $reportSatisfactionSummaries
+     * @return Spreadsheet
+     * @throws \PhpOffice\PhpSpreadsheet\Exception
+     */
+    public function createSatisfactionSpreadsheet($reportSatisfactionSummaries)
+    {
+        $spreadsheet = new Spreadsheet();
+        $spreadsheet->getProperties()
+        ->setCreator("Digideps")
+        ->setLastModifiedBy("Digideps Application")
+        ->setTitle("Satisfaction Report")
+        ->setSubject("Satisfaction Report")
+        ->setDescription(
+            "Output of a particular date range of satisfaction entries."
+        )
+        ->setKeywords("Openxml php")
+        ->setCategory("Satisfaction report results file");
+        $spreadsheet->getDefaultStyle()->getFont()->setName('Arial');
+        $spreadsheet->getDefaultStyle()->getFont()->setSize(12);
+        $sheet = $spreadsheet->getActiveSheet();
+        $sheet->setCellValue('A1', 'ID');
+        $sheet->setCellValue('B1', 'Score');
+        $sheet->setCellValue('C1', 'Created Date');
+        $sheet->setCellValue('D1', 'Comments');
+        $sheet->getColumnDimension('A')->setWidth(10);
+        $sheet->getColumnDimension('B')->setWidth(10);
+        $sheet->getColumnDimension('C')->setWidth(25);
+        $sheet->getColumnDimension('D')->setWidth(70);
+        $headerRows = 'A1:D1';
+        $sheet->getStyle($headerRows)->getFill()->setFillType(Fill::FILL_SOLID)->getStartColor()->setARGB('525252');
+        $sheet->getStyle($headerRows)->getFont()->getColor()->setARGB(Color::COLOR_WHITE);
+        $sheet->getStyle($headerRows)->getFont()->setBold(true);
+
+        $currentCell = 2;
+        foreach ($reportSatisfactionSummaries as $reportSubmissionSummary) {
+            $sheet->setCellValue("A{$currentCell}", $reportSubmissionSummary->getId());
+            $sheet->setCellValue("B{$currentCell}", $reportSubmissionSummary->getScore());
+            $sheet->setCellValue("C{$currentCell}", $reportSubmissionSummary->getCreated());
+            $sheet->getStyle("C{$currentCell}")
+                ->getNumberFormat()
+                ->setFormatCode(NumberFormat::FORMAT_DATE_YYYYMMDDSLASH);
+            $sheet->setCellValue("D{$currentCell}", $reportSubmissionSummary->getComments());
+            $currentCell++;
+        }
+        $sheet->getStyle("A1:D{$currentCell}")
+            ->getAlignment()->setWrapText(true);
+
+        $sheet->setAutoFilter("B1:B{$currentCell}");
+
+        return $spreadsheet;
```

Comment:
Slight overkill using this package - as above, Symfony has this natively supported. You can just throw an array with keys and values and it spits out a csv - Symfony handles this natively with its Serializer - https://symfony.com/blog/new-in-symfony-3-2-csv-and-yaml-encoders-for-serializer

Reply:
It means we can produce prettier spreadsheets and can do things in the future like put in graphs etc. I think it will be pretty nifty for application analytics output.